cooperator/tests/test_emc.py

```diff
@@ -162,30 +162,30 @@ def test_user_rights(self):
         share_type.list_price = 30
         share_type.unlink()
 
-    def test_compute_validated_lines_on_subscription_request(self):
+    def test_compute_is_valid_iban_on_subscription_request(self):
         request = self.request
         request.iban = False
-        request.skip_control_ng = False
-
-        self.assertFalse(request.skip_control_ng)
-        self.assertFalse(request.iban)
+        request.skip_iban_control = False
 
         # empty iban - don't skip
-        self.assertFalse(request.validated)
+        self.assertFalse(request.is_valid_iban)
 
         # good iban - don't skip
-        request.skip_control_ng = False
         self.request.iban = "BE71096123456769"
-        self.assertTrue(request.validated)
+        self.assertTrue(request.is_valid_iban)
 
         # wrong iban - don't skip
         self.request.iban = "xxxx"
-        self.assertFalse(request.validated)
+        self.assertFalse(request.is_valid_iban)
+
+        # wrong iban - don't skip
+        self.request.iban = "BE71096123456760gs"
```

Comment:
```suggestion
         self.request.iban = "BE81096123456769"
```

The first two digits in an IBAN are control digits (ECC). They're like a checksum for all the other digits. If we change a control digit, the IBAN verification should fail.

Or is this check more elementary? Does it support Dutch-style IBANs with text in them? e.g. 'NL91ABNA0417164300'? What happens if the IBAN is spaced out à la 'NL91 ABNA 0417 1643 00'?

Reply:
The `gs` is a typo, I removed it. I intended to fail the checksum by replacing the trailing 9 to 0 and it works. 
The check is actually performed by Odoo Standard so I don't think we should go further in the unit tests.

Reply:
Ah, got it!